### PR TITLE
Add Node CPU/Memory Graphs

### PR DESCRIPTION
**Pull request by notAreYouScared**
Adds an overview tab to EditNode, Shows basic system information, and current host node cpu/memory usage, updated every 5 seconds 

image

### CODE REVIEW

**Comment by notAreYouScared** [comment]
Newest Commit 

image

File: app/Filament/Resources/NodeResource/Widgets/NodeCpuChart.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    protected function getData(): array
+    {
+        /** @var Node $node */
+        $node = $this->record;
+
+        $cpu = collect(cache()->get("nodes.$node->id.cpu_percent"))
+            ->slice(-10)
+            ->map(fn ($value, $key) => [
+                'cpu' => $value,
+                'timestamp' => Carbon::createFromTimestamp($key)->format('H:i:s'),
```
Comment (Boy132) [inline app/Filament/Resources/NodeResource/Widgets/NodeCpuChart.php:27]: Is the time displayed in the user timezone? Or app timezone (UTC)?
This looks like app timezone.

File: app/Filament/Resources/NodeResource/Widgets/NodeMemoryChart.php [hunk 2]
```diff
@@ -3,66 +3,78 @@
...
     {
         /** @var Node $node */
         $node = $this->record;
 
-        $total = ($node->statistics()['memory_total'] ?? 0) / 1024 / 1024 / 1024;
-        $used = ($node->statistics()['memory_used'] ?? 0) / 1024 / 1024 / 1024;
-        $unused = $total - $used;
+        $memUsed = collect(cache()->get("nodes.$node->id.memory_used"))->slice(-10)
+            ->map(fn ($value, $key) => [
+                'memory' => $value / 1024 / 1024 / 1024,
```
Comment (Boy132) [inline app/Filament/Resources/NodeResource/Widgets/NodeMemoryChart.php:25]: All values should be converted to/ from GB/ GiB based on the config value for the binary prefix. (`config('panel.use_binary_prefix')`)

Comment (notAreYouScared) [reply app/Filament/Resources/NodeResource/Widgets/NodeMemoryChart.php:25]: Why even give the option... Seems soo extra.

Comment (Boy132) [reply app/Filament/Resources/NodeResource/Widgets/NodeMemoryChart.php:25]: Because most people want to use MB/ GB. They don't want MiB/ GiB.

Comment (notAreYouScared) [reply app/Filament/Resources/NodeResource/Widgets/NodeMemoryChart.php:25]: Isn't MegaByte, MB. 1024? That's what I'm using? I think our plan was to always use MB/GB

Comment (Boy132) [reply app/Filament/Resources/NodeResource/Widgets/NodeMemoryChart.php:25]: MegaByte = MB = 1000 KB (decimal prefix)
MebiByte = MiB = 1024 KiB (binary prefix)

This is why there is a config option. It's very easy to mess these up.

Comment (notAreYouScared) [reply app/Filament/Resources/NodeResource/Widgets/NodeMemoryChart.php]: The latest commit should correctly use the config variable.

File: app/Filament/Resources/NodeResource/Widgets/NodeCpuChart.php (hunk 1 above)
Comment (notAreYouScared) [reply app/Filament/Resources/NodeResource/Widgets/NodeCpuChart.php]: Set to use the users timezone, if that for some reason does not exist it falls back to UTC

**Review by Boy132 — COMMENTED** [review]
Timezone and binary prefix stuff looks good.

But should we add a little note that in the cpu graph 100% doesn't mean 1 thread? Otherwise it might be confusing/ inconsistent.
Or we divide it by the threads count?

**Comment by notAreYouScared** [comment]
> Timezone and binary prefix stuff looks good.
> 
> But should we add a little note that in the cpu graph 100% doesn't mean 1 thread? Otherwise it might be confusing/ inconsistent. Or we divide it by the threads count?

It's returned from wings as 100% is the cpu as a whole. 

@iamkubi could we change it ?

**Review by Boy132 — APPROVED** [review]
LGTM

**Pull request merged**